main.scss

```diff
@@ -1,21 +1,47 @@
 @import 'o-buttons/main';
 @import 'o-colors/main';
+@import 'o-grid/main';
 @import 'o-icons/main';
 @import 'o-typography/main';
 
 @import './src/scss/color-use-cases';
-@import './src/scss/deprecated';
 @import './src/scss/brand';
 @import './src/scss/mixins';
 @import './src/scss/variables';
 
-@import './src/scss/shared/mixins';
-@import './src/scss/shared/placeholders';
-@import './src/scss/shared/status';
+@import './src/scss/base';
+@import './src/scss/state';
 
-@import './src/scss/alert/main';
-@import './src/scss/notice/main';
-@import './src/scss/action/main';
+/// Output all oMessage features
+/// @param {Map} $opts ['types': ('alert', 'notice', 'action), 'inner': true, 'states': ('error', 'success', 'inform', 'neutral', 'inform-inverse', 'warning', 'warning-light')]
+
+@mixin oMessage ($opts: (
+	'type': ('action', 'alert', 'notice'),
+	'inner': true,
+	'states': ('error', 'success', 'neutral', 'inform', 'inform-inverse', 'warning', 'warning-light')
+)) {
+	$types: map-get($opts, 'type');
+	$states: map-get($opts, 'states');
+	$inner: map-get($opts, 'inner');
```

Comment:
Thinking about the way you wrote about the distinction between type/layout/states in the PR description, perhaps `'layouts': ('inner')` would be more adaptable to future changes?